Overly strict PTY WebSocket URL validation rejects legitimate Websh server connections

## Description

`validateWebSocketURL` in `pkg/runner/pty.go` compares the PTY WebSocket URL's hostname and port against the `server.url` in `alpamon.conf`, rejecting connections when they don't match exactly.

The Websh (PTY WS) proxy server runs on the **same hostname** as the Alpacon API server but on a **different port** in local development. This causes PTY session initialization to always fail.

## Steps to Reproduce

1. Configure `alpamon.conf` with `url = http://localhost:8000`
2. Run alpamon and connect to the Alpacon console
3. Open a PTY terminal session from the console

## Expected Behavior

PTY WebSocket connection should be established successfully, as the Websh proxy server is a legitimate service on the same host.

## Actual Behavior

PTY session initialization fails immediately with:
```
[ERROR] Failed to initialize PTY session. error="WebSocket URL port \"8080\" does not match server port \"8000\""
```

## Cause

Port validation was added in commit `e0584fc` (2026-03-07) as part of Copilot review hardening on top of the original CodeQL SSRF fix.

```go
// pkg/runner/pty.go - validateWebSocketURL()

// Port validation — fails when ports differ; guesses defaults (443/80) when not specified
parsedPort := normalizePort(parsed.Scheme, parsed.Port())
serverPort := normalizePort(serverURL.Scheme, serverURL.Port())
if parsedPort != "" && serverPort != "" && parsedPort != serverPort {
    return fmt.Errorf(...)
}
```

## Infrastructure

| Environment | Alpacon API | Websh (PTY WS) |
|-------------|-------------|-----------------|
| Production | `https://workspace.region.alpacon.io` | `wss://workspace.region.alpacon.io/ws/websh/...` (same host, different path) |
| Local dev | `http://localhost:8000` | `ws://localhost:8080` (same host, different port) |

- The PTY WebSocket URL is provided by the authenticated Alpacon console as part of the `openpty` command
- Production: same hostname, routed via LB/ingress to the proxy server
- Local dev: same hostname (localhost), different port (API=8000, proxy=8080)

## Proposed Fix

- **Keep scheme validation**: Derive expected WS scheme from server URL (`http` → `ws`, `https` → `wss`)
- **Keep hostname validation**: Exact hostname match (same host in both production and local dev)
- **Remove port validation**: Port differs in local dev; guessing default ports (443/80) is unreliable
- **Remove `normalizePort` helper**: No longer needed

--- a/pkg/runner/ftp_test.go
+++ b/pkg/runner/ftp_test.go
@@ -200,11 +200,6 @@ func TestValidateWebSocketURL(t *testing.T) {
 			url:  "wss://console.example.com:443/ws/channel/123",
 		},
 		{
-			name:    "mismatched port",
-			url:     "wss://console.example.com:8080/ws/channel/123",
-			wantErr: true,
-		},
-		{
 			name:    "empty scheme",
 			url:     "console.example.com/ws/channel/123",
 			wantErr: true,
@@ -235,67 +230,3 @@ func TestValidateWebSocketURL_InvalidServerURL(t *testing.T) {
 	}
 }
 
-func TestValidateWebSocketURL_ServerWithExplicitPort(t *testing.T) {
-	prevServerURL := config.GlobalSettings.ServerURL
-	t.Cleanup(func() { config.GlobalSettings.ServerURL = prevServerURL })
-	config.GlobalSettings.ServerURL = "https://console.example.com:8443"
-
-	tests := []struct {
-		name    string
-		url     string
-		wantErr bool
-	}{
-		{
-			name: "matching explicit port",
-			url:  "wss://console.example.com:8443/ws/channel/123",
-		},
-		{
-			name:    "default port does not match explicit",
-			url:     "wss://console.example.com:443/ws/channel/123",
-			wantErr: true,
-		},
-		{
-			name:    "no port does not match explicit",
-			url:     "wss://console.example.com/ws/channel/123",
-			wantErr: true,
-		},
-	}
-
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			err := validateWebSocketURL(tc.url)
-			if tc.wantErr && err == nil {
-				t.Fatalf("validateWebSocketURL(%q) expected error", tc.url)
-			}
-			if !tc.wantErr && err != nil {
-				t.Fatalf("validateWebSocketURL(%q) unexpected error: %v", tc.url, err)
-			}
-		})
-	}
-}
-
-func TestNormalizePort(t *testing.T) {
-	tests := []struct {
-		name   string
-		scheme string
-		port   string
-		want   string
-	}{
-		{"explicit port returned as-is", "wss", "8080", "8080"},
-		{"wss default", "wss", "", "443"},
-		{"ws default", "ws", "", "80"},
-		{"https default", "https", "", "443"},
-		{"http default", "http", "", "80"},
-		{"unknown scheme empty port", "ftp", "", ""},
-		{"case insensitive scheme", "WSS", "", "443"},
-	}
-
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			got := normalizePort(tc.scheme, tc.port)
-			if got != tc.want {
-				t.Fatalf("normalizePort(%q, %q) = %q, want %q", tc.scheme, tc.port, got, tc.want)
-			}
-		})
-	}
-}

--- a/pkg/runner/ftp_test.go
+++ b/pkg/runner/ftp_test.go
@@ -230,3 +230,45 @@ func TestValidateWebSocketURL_InvalidServerURL(t *testing.T) {
 	}
 }
 
+func TestValidateWebSocketURL_ServerWithExplicitPort(t *testing.T) {
+	prevServerURL := config.GlobalSettings.ServerURL
+	t.Cleanup(func() { config.GlobalSettings.ServerURL = prevServerURL })
+	config.GlobalSettings.ServerURL = "https://console.example.com:8443"
+
+	tests := []struct {
+		name    string
+		url     string
+		wantErr bool
+	}{
+		{
+			name: "same port allowed",
+			url:  "wss://console.example.com:8443/ws/channel/123",
+		},
+		{
+			name: "different port allowed",
+			url:  "wss://console.example.com:9090/ws/channel/123",
+		},
+		{
+			name: "no port allowed",
+			url:  "wss://console.example.com/ws/channel/123",
+		},
+		{
+			name:    "different host rejected",
+			url:     "wss://evil.com:8443/ws/channel/123",
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := validateWebSocketURL(tc.url)
+			if tc.wantErr && err == nil {
+				t.Fatalf("validateWebSocketURL(%q) expected error", tc.url)
+			}
+			if !tc.wantErr && err != nil {
+				t.Fatalf("validateWebSocketURL(%q) unexpected error: %v", tc.url, err)
+			}
+		})
+	}
+}
+

--- a/pkg/runner/pty.go
+++ b/pkg/runner/pty.go
@@ -433,28 +433,7 @@ func validateWebSocketURL(rawURL string) error {
 		return fmt.Errorf("WebSocket URL host %q does not match server host %q", parsed.Hostname(), serverURL.Hostname())
 	}
 
-	parsedPort := normalizePort(parsed.Scheme, parsed.Port())
-	serverPort := normalizePort(serverURL.Scheme, serverURL.Port())
-	if parsedPort != "" && serverPort != "" && parsedPort != serverPort {
-		return fmt.Errorf("WebSocket URL port %q does not match server port %q", parsedPort, serverPort)
-	}
-
 	return nil
-}
-
-// normalizePort returns the effective port for a given scheme and port string.
-func normalizePort(scheme, port string) string {
-	if port != "" {
-		return port
-	}
-	switch strings.ToLower(scheme) {
-	case "http", "ws":
-		return "80"
-	case "https", "wss":
-		return "443"
-	default:
-		return ""
-	}
 }
 
 // getPtyUserAndEnv retrieves user information and sets environment variables.

